R/data.table.R

```diff
@@ -746,7 +747,7 @@ replace_dot_alias = function(e) {
           if (length(backtick_idx)) bysub[backtick_idx] = paste0("`",bysub[backtick_idx],"`")
           backslash_idx = grep("\\", bysub, fixed = TRUE)
           if (length(backslash_idx)) bysub[backslash_idx] = gsub('\\', '\\\\', bysub[backslash_idx], fixed = TRUE)
-          bysub = parse(text=paste0("list(",paste(bysub,collapse=","),")"))[[1L]]
+          bysub = parse(text=paste0("list(",paste(bysub,collapse=","),")"))[[1L]] # TODO: use str2lang after R 3.6.0
```

Comment:
why not just
```r
bysub=as.call(c(list(quote(list)), lapply(bysub, as.name)))
```
assuming `bysub` is character at this point

Reply:
looks great but there's use cases like test 573 that we cover so needs a bit more care:

```
DT = data.table("a "=1:2, "b"=3:4," b"=5:6, v=1:6)
test(571, DT[,sum(v),by="b, b"], data.table("b"=3:4, " b"=5:6, V1=c(9L,12L)))
test(572, DT[,sum(v),by="a , b"], data.table("a "=1:2, " b"=5:6, V1=c(9L,12L)))
test(573, DT[,sum(v),by="b, a"], error="object ' a' not found")
```

Reply:
nvm that's already handled with `strsplit` a few lines before